### ISSUE TITLE
ghc `stack build` not working for `concur-vdom`

### ISSUE THREAD

**Issue by mpdairy** [ajnsit/concur#4]
The ghcjs version compiles perfectly, but now I'm trying to get the regular ghc version compiling so I can use it with ghci in emacs. `concure-core` compiles with ghc, but `concur-vdom` has this error:
```
Warning: Directory listed in concur-vdom.cabal file does not exist: ghc-src

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for concur-vdom-0.1.0.0:
    concur-core must match -any, but the stack configuration has no specified version
    jsaddle-webkit2gtk must match -any, but the stack configuration has no specified version
                       (latest applicable is 0.9.4.0)
needed since concur-vdom-0.1.0.0 is a build target.

Recommended action: try adding the following to your extra-deps in /home/teddy/haskell/superpowerscorp/concur/concur-vdom/stack.yaml:
- jsaddle-webkit2gtk-0.9.4.0

You may also want to try the 'stack solver' command
Plan construction failed.
```
I got a little further by adding jsaddle-webkit2gtk as well as including the `concur-core` directory in `stack.yaml`, but then it spits out a bunch of other dependency problems.

**Comment by ajnsit** [comment]
Yeah the ghc build needs a bunch of work to stub out things. I will put it on my todo list.

**Comment by ajnsit** [comment]
I pushed some commits to enable building Concur-React with GHC. To try it out, just run `stack build` in concur-react or concur-react-components/sortable-tree.

Concur-VDOM is lower in my list because that will require changes to the ghcjs-vdom and ghcjs-ffiqq libraries, but I'll get around to it eventually.

**Comment by mpdairy** [comment]
Ok, thanks. I got concur-react to build with ghc and then started converted my project to it, but the -react version has fewer functions and examples, so for now I'll just try to build my little dashboard with the -vdom version and see how it goes.

**Comment by mpdairy** [comment]
Hey, just wanted to let you know that I used Concur (vdom) for a few hours and I really love it so far. I made a list of expandable items (with new IO queries that load their info when they are clicked to expand, and back buttons), only using monadic flow. It was really fun and concise.

**Comment by ajnsit** [comment]
That's good to hear! 👍 Do let me know if something turns out to be cumbersome or not work as expected. I'll be happy to refine the API to suit the most common use cases!

**Comment by robinvd** [comment]
@mpdairy is servant working with ghcjs?

**Comment by mpdairy** [comment]
@robinvd Yep, you just have to move your servant api definition and proxy into some common place (and use `Data.Proxy(Proxy)` instead of Servant-server's `Proxy`), then use this library: https://github.com/plow-technologies/ghcjs-servant-client to generate the functions that call the api. I also used `Network.Wai.Middleware.Cors` on the server to avoid cross origin errors while I'm developing.
